content/articles/statistical-hypothesis-testing-python-implementation/index.md

```diff
@@ -0,0 +1,573 @@
+
+We also use the alternate hypothesis when variable A depends on variable B. 
+
+The formula for calculating Chi-square test is shown below:
+ 
+![chi_square_formula](/engineering-education/statistical-hypothesis-testing-python-implementation/chi-square.png)
+
+Consider the following example:
+ 
+A mathematics teacher conducted a test for a class of 20 students. He expected 5 to score good marks, 7 to score average marks while 8 to fail. Eventually, when the result came out, 5 failed, 6 were average while 9 scored good marks. 
+
+We can use a Chi-square test to analyze the performance of this class.
+ 
+|  | Good  |  Average  | Fail | Total |
+|----|--------|:----------|--------------|--------|
+| Observed| 9 | 6 | 5 | 20 |
+| Expected| 5 | 7 | 8 | 20 |
+
+From the table above, students fall into three categories ranging from what is expected to what is finally recorded. The main categories are `good`, `average`, and `fail`. 
+
+Let's denote the expected and observed values with letters to make it more understandable.
+
+Let the: 
+- Observed good be `A` and the expected good be `a`.
+- Observed average be `B` and the expected average be `b`.
+- Observed fail be `C` and the expected to fail be `c`.
+
+We then find the difference between the expected and observed values, as shown below:
+  
+```
+A-a = 4
+B-b = -1
+C-c = -3
+```
+
+```
+/A-a/² = 16
+/B-b/² = 1
+/C-c/² = 9
+```
+```
+Good = (16/5) = 3.2
+Average = (1/7) = 0.143
+Fail = (9/8) = 1.125
+ 
+Chi-square = 3.2 + 0.143 + 1.125
+Chi-square = 4.468
+```
+For huge datasets, mathematical calculations cannot be done by hand. The implementation of the Chi-square test in Python is discussed in the next section.
+
+#### Chi-square test with Python implementation.
+We need to import `pandas`, `numpy`, `scipy`and , as demonstrated below:
+ 
+```py
+import pandas as pd
+import numpy as np
+import scipy.stats as stats
+from scipy.stats import f_oneway
+from scipy.stats import chi2_contingency
+import matplotlib.pyplot as plt
+ 
+import statsmodels.api as sm
+from statsmodels.formula.api import ols
+ 
+import warnings
+warnings.filterwarnings( "ignore" )
+```
+ 
+The libraries imported are the most important libraries used for any data analysis project.
+
+`f_oneway` and `chi2_contigency` libraries are used for implementing a one-way `ANOVA` and Chi-square tests respectively.
+ 
+In this tutorial, we will be using the popular iris dataset. You can download it from [here](https://www.kaggle.com/vikrishnan/iris-dataset).
+ 
+The next step is to load the data into the notebook:
+
+```py
+df = pd.read_csv("/content/drive/MyDrive/IRIS.csv")
+df.columns =['sepal_length', 'sepal_width', 'petal_length', 'petal_width', 'species']
+```
+
+```py
+df.head()
+```
+
+![dataset head](/engineering-education/statistical-hypothesis-testing-python-implementation/dataset_head.png)
+
+let's run df.sample() to visualize samples randomly.
```

Comment:
Capitalize the "l" in "let's", and `df.sample()` must be in the format mentioned in this comment (within inverted quotes)